docs/wiki/concepts/CSP-Node.md

```diff
@@ -0,0 +1,276 @@
+
+- **`tickedvalues`**: iterator of values of all ticked inputs
+- **`tickedkeys`**: iterator of keys of all ticked inputs (keys are list index for list baskets)
+- **`tickeditems`**: iterator of (key,value) tuples of ticked inputs
+- **`validvalues`**: iterator of values of all valid inputs
+- **`validkeys`**: iterator of keys of all valid inputs
+- **`validitems`**: iterator of (key,value) tuples of valid inputs
+- **`keys`**: list of keys on the basket (**dictionary baskets only** )
+
+10-11) This demonstrates the ability to access an individual element of a
+basket and react to it as well as access its current value
+
+## **Node Outputs**
+
+Nodes can return any number of outputs (including no outputs, in which case it is considered an "output" or sink node,
+see [Graph Pruning](https://github.com/Point72/csp/wiki/0.-Introduction#graph-pruning)).
+Nodes with single outputs can return the output as an unnamed output.
+Nodes returning multiple outputs must have them be named.
+When a node is called at graph building time, if its is a single unnamed node the return variable is an edge representing the output which can be passed into other nodes.
+If the outputs are named, the return value is an object with the outputs available as attributes.
+For example (examples below demonstrate various ways to output the data as well)
+
+```python
+@csp.node
+def single_unnamed_outputs(n: ts[int]) -> ts[int]:
+    # can either do
+    return n
+    # or
+    # csp.output(n) to continue processes after output
+
+
+@csp.node
+def multiple_named_outputs(n: ts[int]) -> csp.Outputs(y=ts[int], z=ts[float]):
+    # can do
+    # csp.output(y=n, z=n+1.) to output to multiple outputs
+    # or separate the outputs to tick out at separate points:
+    # csp.output(y=n)
+    # ...
+    # csp.output(z=n+1.)
+    # or can return multiple values with:
+    return csp.output(y=n, z=n+1.)
+
+@csp.graph
+def my_graph(n: ts[int]):
+    x = single_unnamed_outputs(n)
+    # x represents the output edge of single_unnamed_outputs,
+    # we can pass it a time series input to other nodes
+    csp.print('x', x)
+
+
+    result = multiple_named_outputs(n)
+    # result holds all the outputs of multiple_named_outputs, which can be accessed as attributes
+    csp.print('y', result.y)
+    csp.print('z', result.z)
+```
+
+## Basket Outputs
+
+Similarly to inputs, a node can also produce a basket of timeseries as an output.
+For example:
+
+```python
+class MyStruct(csp.Struct):            # 1
+    symbol: str                        # 2
+    index: int                         # 3
+    value: float                       # 4
+                                       # 5
+@csp.node                              # 6
+def demo_basket_output_node(           # 7
+    in_: ts[MyStruct],                 # 8
+    symbols: [str],                    # 9
+    num_symbols: int                   # 10
+) -> csp.Outputs(                      # 11
+    dict_basket=csp.OutputBasket(      # 12
```

Comment:
maybe we can widen the line length on these so the signature isnt spread across so many lines, makes it harder to digest

Reply:
I see Melissa updated this, resolving